crates/sui-kvstore/src/bigtable/client.rs

```diff
@@ -0,0 +1,411 @@
+                        } else if chunk.timestamp_micros >= timestamp {
+                            // newer version of cell is available
+                            timestamp = chunk.timestamp_micros;
+                            cell_value = chunk.value;
+                        }
+                    }
+                }
+                if chunk.row_status.is_some() {
+                    if let Some(RowStatus::CommitRow(_)) = chunk.row_status {
+                        if let Some(cell_name) = cell_name {
+                            row.push((cell_name, cell_value));
+                        }
+                        if let Some(row_key) = row_key {
+                            result.push((row_key, row));
+                        }
+                    }
+                    row_key = None;
+                    row = vec![];
+                    cell_value = vec![];
+                    cell_name = None;
+                }
+            }
+        }
+        Ok(result)
+    }
+
+    async fn multi_set(
+        &mut self,
+        table_name: &str,
+        values: impl IntoIterator<Item = (Bytes, Vec<(&str, Bytes)>)> + std::marker::Send,
+    ) -> Result<()> {
+        let mut entries = vec![];
+        for (row_key, cells) in values {
+            let mutations = cells
+                .into_iter()
+                .map(|(column_name, value)| Mutation {
+                    mutation: Some(mutation::Mutation::SetCell(SetCell {
+                        family_name: COLUMN_FAMILY_NAME.to_string(),
+                        column_qualifier: column_name.to_owned().into_bytes(),
+                        timestamp_micros: -1,
+                        value,
+                    })),
+                })
+                .collect();
+            entries.push(Entry { row_key, mutations });
+        }
+        let request = MutateRowsRequest {
+            table_name: format!("{}{}", self.table_prefix, table_name),
+            entries,
+            ..MutateRowsRequest::default()
+        };
+        self.mutate_rows(request).await?;
+        Ok(())
+    }
+
+    pub async fn multi_get(
+        &mut self,
+        table_name: &str,
+        keys: Vec<Vec<u8>>,
+    ) -> Result<Vec<Vec<(Bytes, Bytes)>>> {
+        let request = ReadRowsRequest {
+            table_name: format!("{}{}", self.table_prefix, table_name),
+            rows_limit: keys.len() as i64,
+            rows: Some(RowSet {
+                row_keys: keys,
+                row_ranges: vec![],
+            }),
+            ..ReadRowsRequest::default()
+        };
+        let mut result = vec![];
+        for (_, cells) in self.read_rows(request).await? {
+            result.push(cells);
+        }
+        Ok(result)
+    }
+
+    fn raw_object_key(object_key: &ObjectKey) -> Result<Vec<u8>> {
+        let mut raw_key = bcs::to_bytes(&object_key.0)?;
+        raw_key.extend(object_key.1.value().to_be_bytes());
+        Ok(raw_key)
```

Comment:
Lets make this more explicit, and we can completely avoid the the bcs logic and make this infallible. You can convert the ObjectID to a Vec and then push the be bytes of the version onto the vec